### ISSUE TITLE
Migrating to Bootstrap v4

### ISSUE THREAD

**Issue by cleverbeagle** [cleverbeagle/pup#122]
Now that Bootstrap v4 is out, it's worth discussing a migration to something like [reactstrap](https://reactstrap.github.io/). The only hesitation here is that it'd be a decent shift in APIs that we've all grown accustom to. It _does_ seem like the React Bootstrap folks are working on a v4 update, but progress is slow. I'm indifferent as the changes are mostly visual from what I can see (a few new components but otherwise nothing that's an emergency).

**Comment by cleverbeagle** [comment]
@denodaeus no ETA. We're at the mercy of the `react-bootstrap` folks. The reason we're not making a jump to `reactstrap` is two fold:

1.) It's a different API and our goal with Pup is to service our mentorship clients (and ourselves while teaching them) by keeping things as consistent as possible. If we can avoid the need to memorize a fresh API, it's worth waiting.
2.) Because it's mostly visual changes in BS4, it's not a high priority to shift over—the bang for buck value is low enough to be patient with the `react-bootstrap` folks.

Ultimately, this is why we're working on [Fur](https://cleverbeagle.com/fur)—we want to remove any unnecessary stagnation/barricades to innovation in respect to the UI.

**Comment by cleverbeagle** [comment]
Going to close this one. It would be best to just focus on migrating Pup to use [Fur](https://cleverbeagle.com/fur) moving forward so we can avoid headaches with this.

**Comment by cleverbeagle** [comment]
Re-opening this. I'd still like Fur to take over but realistically that's going to take some time. To avoid confusion and headaches, it's best to just migrate to Bootstrap v4.

**Comment by wortkotze** [comment]
Just wanted to ask if there is any ETA for a change to bootstrap 4?

**Comment by cleverbeagle** [comment]
@wortkotze officially, no, but it's on my radar. Is it something you'd be open to helping with?

**Comment by wortkotze** [comment]
I have a branch that’s already working on b4 but it seems to have a little bit more dependencies that we have to update. 
What is your latest dev branch ? When I have some time I will see what I can do as it’s getting more and more complicated to get support for b3

**Comment by wortkotze** [comment]
I checked it tonight, the most issues are coming from SSR. any plans to work on this that it works with never react and react-dom versions?

**Comment by wortkotze** [comment]
I will try it out tonight. If someone has some infos how u want to build ssr including black and white list I.e via prop of the route etc would be helpfull

**Comment by wortkotze** [comment]
I migrated the actual pub version now tp bootstrap4 could u give me an branch where I can push it?

**Comment by cleverbeagle** [comment]
Hey @wortkotze, the best way to share your work is to submit a pull request to this repository. Instructions here: https://help.github.com/en/articles/creating-a-pull-request

**Comment by wortkotze** [comment]
I would but I cant open a new branch ?

**Comment by wortkotze** [comment]
[PR319](https://github.com/cleverbeagle/pup/pull/319)

**Comment by pilarArr** [comment]
Hey @wortkotze, I've seen your pull request for the migration to bootstrap 4, and I saw that there were a few things leftover from bootstrap 3, so I gave it a go at the migration. I'm going to create a pull request but I don't want to be overstepping anyone, so anyone feel free to discard this if I'm overstepping.
I'm going to tinker with React Hooks in Pup, but I just wanted to get the migration done before going in.
